certbot/tests/ocsp_test.py

```diff
@@ -139,116 +140,131 @@ def test_ensure_cryptography_toggled(self, mock_revoke, mock_determine):
+        self.assertFalse(revoked)
 
+        # OSCP signature response is invalid.
+        with _ocsp_mock(ocsp_lib.OCSPCertStatus.REVOKED, ocsp_lib.OCSPResponseStatus.SUCCESSFUL,
+                        check_signature_side_effect=InvalidSignature('foo')):
+            revoked = self.checker.ocsp_revoked(self.cert_path, self.chain_path)
         self.assertFalse(revoked)
 
-        # Valid response, OCSP extension is present,
-        # but OCSP response uses an unsupported signature.
-        mock_ocsp_response.return_value = _construct_mock_ocsp_response(
-            ocsp_lib.OCSPCertStatus.REVOKED, ocsp_lib.OCSPResponseStatus.SUCCESSFUL)
-        mock_post.return_value = mock.Mock(status_code=200)
-        mock_check.side_effect = UnsupportedAlgorithm('foo')
-        revoked = self.checker.ocsp_revoked(self.cert_path, self.chain_path)
+        # Assertion error on OCSP response validity
+        with _ocsp_mock(ocsp_lib.OCSPCertStatus.REVOKED, ocsp_lib.OCSPResponseStatus.SUCCESSFUL,
+                        check_signature_side_effect=AssertionError('foo')):
+            revoked = self.checker.ocsp_revoked(self.cert_path, self.chain_path)
+        self.assertFalse(revoked)
 
+        # No responder cert in OCSP response
+        with _ocsp_mock(ocsp_lib.OCSPCertStatus.REVOKED,
+                        ocsp_lib.OCSPResponseStatus.SUCCESSFUL) as mocks:
+            mocks['mock_response'].return_value.certificates = []
+            revoked = self.checker.ocsp_revoked(self.cert_path, self.chain_path)
         self.assertFalse(revoked)
 
-        # And now, the signature itself is invalid.
-        mock_ocsp_response.return_value = _construct_mock_ocsp_response(
-            ocsp_lib.OCSPCertStatus.REVOKED, ocsp_lib.OCSPResponseStatus.SUCCESSFUL)
-        mock_post.return_value = mock.Mock(status_code=200)
-        mock_check.side_effect = InvalidSignature('foo')
-        revoked = self.checker.ocsp_revoked(self.cert_path, self.chain_path)
+        # Responder cert is not signed by certificate issuer
+        with _ocsp_mock(ocsp_lib.OCSPCertStatus.REVOKED,
+                        ocsp_lib.OCSPResponseStatus.SUCCESSFUL) as mocks:
+            cert = mocks['mock_response'].return_value.certificates[0]
+            mocks['mock_response'].return_value.certificates[0] = mock.Mock(
+                issuer='fake', subject=cert.subject)
+            revoked = self.checker.ocsp_revoked(self.cert_path, self.chain_path)
+        self.assertFalse(revoked)
 
+        with _ocsp_mock(ocsp_lib.OCSPCertStatus.REVOKED, ocsp_lib.OCSPResponseStatus.SUCCESSFUL):
+            # This mock is necessary to avoid the first call contained in _determine_ocsp_server
+            # of the method cryptography.x509.Extensions.get_extension_for_class.
+            with mock.patch('certbot.ocsp._determine_ocsp_server') as mock_server:
+                mock_server.return_value = ('https://i.do.not.care', 'i.do.not.care')
```

Comment:
Even though these don't do anything. I think we should stick with `example.com` domains in here to avoid misunderstandings and such (`not.care` is an actual, registered domain name).

Reply:
My creativity is frustrated! But done, with the usual dns example... :)